transport/http_util.go

```diff
@@ -288,6 +289,80 @@ func timeoutDecode(s string) (time.Duration, error) {
 	return d * time.Duration(t), nil
 }
 
+const (
+	spaceByte   = ' '
+	tildaByte   = '~'
+	percentByte = '%'
+)
+
+// grpcMessageEncode is used to encode status code in header field
+// "grpc-message".
```

Comment:
It's consistent with `timeoutEncode()`.
However, maybe it's better to call it `statusDescEncode`, because it's called as
```go
t.hEnc.WriteField(hpack.HeaderField{Name: "grpc-message", Value: grpcMessageEncode(statusDesc)})
```